pkg/skaffold/sync/sync.go

```diff
@@ -172,6 +176,30 @@ func syncMapForArtifact(a *latest.Artifact, insecureRegistries map[string]bool)
 	}
 }
 
+func autoSyncItem(ctx context.Context, a *latest.Artifact, e filemon.Events, builds []build.Artifact) (*Item, error) {
+	tag := latestTag(a.ImageName, builds)
+	if tag == "" {
+		return nil, fmt.Errorf("could not find latest tag for image %s in builds: %v", a.ImageName, builds)
+	}
+
+	switch {
+	case a.JibArtifact != nil:
+		toCopy, toDelete, err := jib.GetSyncDiff(ctx, a.Workspace, a.JibArtifact, e)
+		if err != nil {
+			return nil, err
+		}
+		if toCopy == nil && toDelete == nil {
+			// do a rebuild
+			return nil, nil
+		}
+		return &Item{Image: tag, Copy: toCopy, Delete: toDelete}, nil
+
+	default:
+		// TODO: this error does appear a little late in the build, perhaps it could surface at first run, rather than first sync?
```

Comment:
agreed, this could be validated upfront - for now this should be fine, that extra couple of seconds is not really worth pulling this thing only to the validation, also we should validate the config consistently then - which we don't do currently.